include/boost/hana/experimental/printable.hpp

```diff
@@ -111,6 +111,18 @@ BOOST_HANA_NAMESPACE_BEGIN namespace experimental {
         }
     };
 
+    // model for hana::optional
+    template <>
+    struct print_impl<hana::optional_tag> {
+        template <typename O>
+        static auto apply(O const& optional) {
+            return hana::maybe("nothing",
+                [](auto x) {
```

Comment:
This will make a copy of `x` because you take by value. Can you please use `auto const& x`?